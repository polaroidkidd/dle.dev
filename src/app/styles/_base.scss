@import "variables";

html {
  font-size: 62.5%; // 1rem = 10px (for default font size of 16px)
}

body {
  background-color: $color-primary-background;
  box-sizing: border-box;
  margin: 0 auto;
  max-width: 140rem;
  padding: 0 1rem;
}