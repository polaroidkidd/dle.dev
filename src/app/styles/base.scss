body {
  background-color: $color-primary-background;
}

h1 {
  color: $color-primary-heading;
}

