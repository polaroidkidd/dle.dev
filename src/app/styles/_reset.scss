*,
*::after,
*::before {
  margin: 0;
  padding: 0;
  box-sizing: inherit;
}

img, embed, object, video {
  max-width: 100%;
}

html {
  font-size: 62.5%; // 1rem = 10px (for default font size of 16px)
}

body {
  box-sizing: border-box;
}