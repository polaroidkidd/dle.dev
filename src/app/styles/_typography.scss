@import "variables";

@mixin heading-primary {
  color: $color-primary-text;
  font-family: Poppins, sans-serif;
  font-size: 5rem;
  font-weight: 300;
  width: 100%;
  @media (max-width: 768px) {
    font-size: 4rem;
  }
}

@mixin heading-secondary {
  color: $color-primary-text;
  font-family: Poppins, sans-serif;
  font-size: 4rem;
  font-weight: 300;
  letter-spacing: 0.2rem;
  margin-right: -2rem;
  margin-top: 3rem;
  width: 100%;
  @media (max-width: 768px) {
    font-size: 3rem;
  }
}

@mixin heading-tertiary {
  color: $color-primary-text;
  font-family: Poppins, sans-serif;
  font-size: 3rem;
  font-weight: 300;
  letter-spacing: 0.75rem;
  @media (max-width: 768px) {
    font-size: 2rem;
  }
}

.heading-primary {
  @include heading-primary;
}

.heading-secondary {
  @include heading-secondary
}


.heading-tertiary {
  @include heading-tertiary
}

@mixin paragraph {
  color: $color-primary-text;
  font-family: Poppins, sans-serif;
  font-size: 1.75rem;
  @media (max-width: 768px) {
    font-size: 2rem;
    font-weight: 200;
  }
}

.paragraph {
  @include paragraph
}

h1, h2, h3, h4, h5, h6, p {
  line-height: 3rem;
}