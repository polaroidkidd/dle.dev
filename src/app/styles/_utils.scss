
.u-center-text {
  &__horizontal {
    text-align: center;
  }

  &__padding {
    padding: 4rem;
  }
}