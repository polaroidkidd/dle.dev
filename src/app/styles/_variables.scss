$color-background-primary: #1A1A1D;
$color-text-primary: #FFFAFE;
$color-primary-border: #4F2626;
$color-primary-border-hover: #D6304C;
$color-text-primary-accent: #DA746F;