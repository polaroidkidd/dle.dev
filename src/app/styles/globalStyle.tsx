import { createGlobalStyle } from 'styled-components';


export const GlobalStyle = createGlobalStyle<{ theme: ThemeType }>`
*,
*::after,
*::before {
  margin: 0;
  padding: 0;
  box-sizing: inherit;
}

img, embed, object, video {
  max-width: 100%;
}

html {
  font-size: 62.5%; // 1rem = 10px (for default font size of 16px)
}

body {
  box-sizing: border-box;
}

/* Coolors Exported Palette - coolors.co/9ad1d4-80ced7-e4dfda-d29593-1a1a1d */

body {
  background-color: ${props => props.theme.primary.colorPrimaryBackground};
}

h1 {
  color: ${props => props.theme.primary.colorPrimaryHeading};
}

`;


export const theme = {
  primary: {
    colorTertiaryHeading: '#9AD1D4',
    colorSecondaryHeading: '#80CED7',
    colorPrimaryText: '#E4DFDA',
    colorPrimaryHeading: '#D29593',
    colorPrimaryBackground: '#1A1A1D',
    colorPrimaryBorder: '#0BD29E',
  }
};

export type ThemeType = typeof theme;
