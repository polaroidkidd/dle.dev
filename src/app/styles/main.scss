@import "reset";
@import "utils";
@import "variables";
@import "typography";
@import "base";