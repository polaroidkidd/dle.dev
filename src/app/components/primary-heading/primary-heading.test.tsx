import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import { PrimaryHeading } from './index';
import { ThemeProvider } from 'styled-components';
import { theme } from '../../styles/globalStyle';


test('render PrimaryHeading with text', () => {
  const tree = renderer.create(
    <ThemeProvider theme={theme}>
      <PrimaryHeading text="test heading"/>
    </ThemeProvider>).toJSON();
  expect(tree).toMatchSnapshot();
});
