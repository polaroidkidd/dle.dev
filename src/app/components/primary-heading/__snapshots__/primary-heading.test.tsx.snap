// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render PrimaryHeading with text 1`] = `
.c0 {
  font-size: 5rem;
  font-family: Poppins,sans-serif;
  font-weight: 100;
  color: #0BD29E;
}

<div
  className="c0"
>
  test heading
</div>
`;
